tests/phpunit/elementor/includes/base/test-widget-base.php

```diff
@@ -31,6 +31,31 @@ public function tearDown() {
 		Plugin::$instance->frontend->render_mode_manager = $this->default_render_mode;
 	}
 
+	public function test_set_config_string() {
+		require_once __DIR__ . '/mock/mock-widget.php';
+
+		$widget = new Mock_Widget( [ 'settings' => [], 'id' => '1' ], [] );
+
+		$widget->set_config( 'test', 'test' );
+
+		$config = $widget->get_config();
+
+		$this->assertEquals( 'test', $config[ 'test' ] );
+	}
+
+	public function test_set_config_array() {
+		require_once __DIR__ . '/mock/mock-widget.php';
+
+		$widget = new Mock_Widget( [ 'settings' => [], 'id' => '1' ], [] );
+
+		$widget->set_config( 'test', [ 'first' ] );
+		$widget->set_config( 'test', [ 'second' ] );
```

Comment:
Please add a test that overrides a value in the array
```
$widget->set_config( 'test', [ 'first' => 1 ] );
$widget->set_config( 'test', [ 'first' => 2 ] );
```

Reply:
The desired behavior is:
`$widget->set_config( 'categories', [ 'favorites' ] ); // [ 'basic', 'favorites' ]`

Which means there should be a merge, not an override.

Reply:
Please add a test that ensure it doesn't override. In order to avoid a future overriding.